From mboxrd@z Thu Jan  1 00:00:00 1970
Return-Path: <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 1CA1BD46BE7
	for <linux-rockchip@archiver.kernel.org>; Wed, 28 Jan 2026 17:49:02 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-ID:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=/Tlf0WkQlpk/OkX4uQSFAAZEL6qgEI7T1vo14ZIXJ0A=; b=g8bXLhPNpbJjwu
	4jtqLDxZrQSBw7yjbcoERweaExc3LP9WXfSCFlIcrXEDNg+fo7XGfB1l2eMarAzYvZR99J5AAD1FG
	0I9O9I0DDHRwC4Py/AumI9XBYxQxwCgze54W7uEm8x4Y+DTVzxhJvR3/Gy6GFo5k3nKES4IeHq0Kn
	dAPWEBEHoPNlYS5ix3gXTHnNJ1AfQJMTR/eNncRELz8kcYBdhp7qLDwMu15GrneuLxq8XHyj8dDNt
	HdPTxaTA+22m9Nhefk/k1EkIPSgiaXtUH1WFz5DQId9+6Ii5oUveKB9LxRLQNJX1lCGr0vPo0B8ib
	txLHvPFJTVrgqaCGWC8w==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.98.2 #2 (Red Hat Linux))
	id 1vl9ep-0000000GWS7-2aq0;
	Wed, 28 Jan 2026 17:48:47 +0000
Received: from mail-oa1-x29.google.com ([2001:4860:4864:20::29])
	by bombadil.infradead.org with esmtps (Exim 4.98.2 #2 (Red Hat Linux))
	id 1vl9en-0000000GWQR-0jhx
	for linux-rockchip@lists.infradead.org;
	Wed, 28 Jan 2026 17:48:46 +0000
Received: by mail-oa1-x29.google.com with SMTP id 586e51a60fabf-4094fbd1808so38109fac.1
        for <linux-rockchip@lists.infradead.org>; Wed, 28 Jan 2026 09:48:44 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20230601; t=1769622524; x=1770227324; darn=lists.infradead.org;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:from:to:cc:subject:date
         :message-id:reply-to;
        bh=Z4g1rdqkzBN7E4k3DRb93C6Pz1pOJ7VGgwv2jByOH00=;
        b=G9Fth0zBNQ0jfmftoF/QgR96OKw9zMqRNZhqxAEENQUqvRTARcrSPjqAg58VjSFD7W
         bWol8zBoXtAQbu8fHQShcCBETN4F7riGxqJHQ48paMlJCWNOGp8sjqN1UtuNOmg8XlR3
         3yG6e0awSSeycUCnlL8HZsdOXlLJPeuEZcNgRnhgsYIcnxz2uPN39/AkqrXLQmorW7ZJ
         4h/76rxyhji3JVHqig7TV7W94XJEH5+sVSsWnqCi1SX0OUFfvgdR7Us6Wrft1kEntk2/
         hDd4tUDvhKzeWe9VLKG6bzs/4ORoAmiq7daqw6YgIb27i+p9l4oGu7wvoTFluFmxNv+K
         ns5Q==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1769622524; x=1770227324;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:x-gm-gg:x-gm-message-state:from
         :to:cc:subject:date:message-id:reply-to;
        bh=Z4g1rdqkzBN7E4k3DRb93C6Pz1pOJ7VGgwv2jByOH00=;
        b=BjCn9Lpp9M6a1ev3sB1iNDsVTg4HkXWGa+ururqqr1XuaAI1LDn5uX/KP5JMfYmP1J
         a+5twQXQ7MMGjHz9PNcixq3bvniBC2AhvV2qbEiVyc6vsXxsPPcxArPNLgITrPCrZQ5Y
         CJlalOAGme/h7cRrVN47tdLDQEu5hVx1WOyqLPoCExnyr8Nl1MF/Z3SQHTHTA77u1NRk
         edGmPe+mpiRMXqM4Xfm6hXHBfatlXDCheG4fg0tHlM8gQXCLAAv17gM+phcBAdhwWMhi
         pL9kFUpQZMW7CJoj0HC2x2JELFeZpVyAz9MzogCvXUtUor6UNeD34/kJBpWcccptzoWV
         +3sA==
X-Forwarded-Encrypted: i=1; AJvYcCWDgFvt12fX/CWDbJopwIzNu8LZQ/kc1dDHVUBrcdz03p/dOEVvkWzl50I1PgjVxIkJAEpu7skpscD54IPVrg==@lists.infradead.org
X-Gm-Message-State: AOJu0YxYj8Vdw2GdRIVzhVZCQ/WE3V6r7YxUpI+91yXG83IcXsJ7da5f
	U7UF6AGLPhTqqW2cjBoJesVxXGOE118ra/oglYCxxE+H2czvtWTEBdn8zeWwNg==
X-Gm-Gg: AZuq6aLQs0+YmVo/vvhO4sJja9QtBbNWRfsdkuUlxhZhP1BGEN8lU5YAwtJCEiqp2AT
	EZpvbzBOhEQbEiZj4DcCJOLAFdTrHRi1T+ckNn2ZFEtwygfbqnwyXoXGCs4312rpwlZihrDN/uc
	JGz4lwL61yCWifHRsbGq6B1qSLtplXVup07lej7c2s+t7zrtQ3G5Ogsdw3gV8/SXhbO/AapaYXq
	ruRBHzHGrG1whljsXtoaauPV1f5zs3un0e1Tit/YHbxkHoEVBYeNIMYKBemdS2GsLp6DjR6a6N8
	E4rukiCPV1MAQWbQyU+vgVTCVLTN/TZ43647u7sVb3ib+fR072GHp7YKxCQSXmMr56OtE5Ln3Ce
	kNRcLg3Nx4DlOihQ6xt4PUm2VLBfurcl7IGQEvJSuVnNibwSuVjyrDp3+bt214t7AVjNq0yPS+X
	sNrFwPG6qK
X-Received: by 2002:a05:6870:b01b:b0:3f6:1e56:aad7 with SMTP id 586e51a60fabf-4093fce5f4emr3436398fac.15.1769622522315;
        Wed, 28 Jan 2026 09:48:42 -0800 (PST)
Received: from localhost.localdomain ([2600:1700:fb0:1bc0::54])
        by smtp.gmail.com with ESMTPSA id 586e51a60fabf-409575b0a89sm2187647fac.20.2026.01.28.09.48.40
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 28 Jan 2026 09:48:41 -0800 (PST)
From: Chris Morgan <macroalpha82@gmail.com>
To: linux-sound@vger.kernel.org
Cc: devicetree@vger.kernel.org,
	wangweidong.a@awinic.com,
	tiwai@suse.com,
	perex@perex.cz,
	conor+dt@kernel.org,
	krzk+dt@kernel.org,
	robh@kernel.org,
	broonie@kernel.org,
	lgirdwood@gmail.com,
	heiko@sntech.de,
	linux-rockchip@lists.infradead.org,
	Chris Morgan <macromorgan@hotmail.com>,
	Conor Dooley <conor.dooley@microchip.com>
Subject: [PATCH V2 1/3] ASoC: dt-bindings: aw87390: Add Anbernic RG-DS Amplifier
Date: Wed, 28 Jan 2026 11:46:06 -0600
Message-ID: <20260128174608.1498-2-macroalpha82@gmail.com>
X-Mailer: git-send-email 2.43.0
In-Reply-To: <20260128174608.1498-1-macroalpha82@gmail.com>
References: <20260128174608.1498-1-macroalpha82@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20260128_094845_218791_7842D33F 
X-CRM114-Status: GOOD (  10.06  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

From: Chris Morgan <macromorgan@hotmail.com>

Add a binding for the Anbernic RG-DS Amplifier, which is an Awinic
aw87391 audio amplifier. This manufacturer did not provide firmware
so we have to use a list of init commands instead, requiring device
specific functionality rather than generic aw87391 functionality.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
Acked-by: Conor Dooley <conor.dooley@microchip.com>
---
 .../bindings/sound/awinic,aw87390.yaml        | 34 ++++++++++++++++---
 1 file changed, 29 insertions(+), 5 deletions(-)

diff --git a/Documentation/devicetree/bindings/sound/awinic,aw87390.yaml b/Documentation/devicetree/bindings/sound/awinic,aw87390.yaml
index ba9d8767c5d5..9c1baae767c4 100644
--- a/Documentation/devicetree/bindings/sound/awinic,aw87390.yaml
+++ b/Documentation/devicetree/bindings/sound/awinic,aw87390.yaml
@@ -15,12 +15,15 @@ description:
   sound quallity, which is a new high efficiency, low
   noise, constant large volume, 6th Smart K audio amplifier.
 
-allOf:
-  - $ref: dai-common.yaml#
-
 properties:
   compatible:
-    const: awinic,aw87390
+    oneOf:
+      - enum:
+          - awinic,aw87390
+      - items:
+          - enum:
+              - anbernic,rgds-amp
+          - const: awinic,aw87391
 
   reg:
     maxItems: 1
@@ -40,10 +43,31 @@ required:
   - compatible
   - reg
   - "#sound-dai-cells"
-  - awinic,audio-channel
 
 unevaluatedProperties: false
 
+allOf:
+  - $ref: dai-common.yaml#
+  - if:
+      properties:
+        compatible:
+          contains:
+            enum:
+              - awinic,aw87390
+    then:
+      required:
+        - awinic,audio-channel
+
+  - if:
+      properties:
+        compatible:
+          contains:
+            enum:
+              - anbernic,rgds-amp
+    then:
+      properties:
+        vdd-supply: true
+
 examples:
   - |
     i2c {
-- 
2.43.0


_______________________________________________
Linux-rockchip mailing list
Linux-rockchip@lists.infradead.org
http://lists.infradead.org/mailman/listinfo/linux-rockchip

